pre_release.sh

```diff
@@ -0,0 +1,71 @@
+#!/bin/bash
+
+usage() { echo "Usage: $0 -t <tag>" 1>&2; exit 1; }
+
+while getopts ":t:" o; do
+    case "${o}" in
+        t)
+            tag=${OPTARG}
+            ;;
+        *)
+            usage
+            ;;
+    esac
+done
+if [ ! -z "${t}" ]; then
+    usage
+fi
+
+#validate tag
+semver_regex="^(0|[1-9][0-9]*)\\.(0|[1-9][0-9]*)\\.(0|[1-9][0-9]*)(\\-[0-9A-Za-z-]+(\\.[0-9A-Za-z-]+)*)?(\\+[0-9A-Za-z-]+(\\.[0-9A-Za-z-]+)*)?$"
+if [[ ${tag} =~ ${semver_regex} ]]; then
+    echo "${tag} is valid semver tag"
+else
+    echo "Error: ${tag} is not a valid semver tag. Exiting"
+    exit -1
+fi
+
+#ensure tag doesn't exits
+if [[ $(git tag --list ${tag}) ]]; then
+   echo "Error: Tag ${tag} already exists. Exiting"
+   exit -1
+fi
+
+if ! git diff --quiet; then \
+	printf "Working tree is not clean, can't proceed with the release process\n"
+	git status
+	git diff
+	exit 1
+fi
+
+# check changelog exists
+changelog_file="./CHANGELOG.md"
```

Comment:
How does manually managing `CHANGELOG.md` interact with built-in GitHub release management tab? Is intention to populate the markdown, then copy individual sections into Release tab?

Reply:
Correct